apis/s3/src/main/java/org/jclouds/s3/domain/ListMultipartUploadResponse.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.s3.domain;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Date;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.annotations.Beta;
+
+@AutoValue
+@Beta
+public abstract class ListMultipartUploadResponse {
+   public abstract int partNumber();
+   public abstract Date lastModified();
+   public abstract String eTag();
+   public abstract long size();
+
+   public static ListMultipartUploadResponse create(int partNumber, Date lastModified, String eTag, long size) {
+      checkArgument(partNumber > 0, "partNumber must be greater than zero, was: " + partNumber);
+      checkNotNull(eTag, "eTag");
+      lastModified = (Date) checkNotNull(lastModified, "lastModified").clone();
+      checkArgument(size >= 0, "size must be positive, was: " + size);
```

Comment:
Done twice.  Some versions of error-prone warn about this but the check was reparented into the Google-only Klippy.